presto-memory/src/main/java/com/facebook/presto/plugin/memory/MemorySplit.java

```diff
@@ -50,7 +51,7 @@ public MemorySplit(
         this.tableHandle = requireNonNull(tableHandle, "tableHandle is null");
         this.partNumber = partNumber;
         this.totalPartsPerWorker = totalPartsPerWorker;
-        this.addresses = ImmutableList.copyOf(requireNonNull(addresses, "addresses is null"));
+        this.addresses = ImmutableList.of(requireNonNull(address, "address is null"));
```

Comment:
I didn't want to create new lists for each `getAddresses` call, but now looking at `SingletonImmutableList` it doesn't seem like an overhead at all.